crypto/x509/x509_cmp.c

```diff
@@ -173,7 +173,7 @@ int X509_NAME_cmp(const X509_NAME *a, const X509_NAME *b)
 
     ret = a->canon_enclen - b->canon_enclen;
 
-    if (ret)
+    if (ret != 0 || a->canon_enclen == 0)
```

Comment:
While we are at it, wouldn't it be appropriate to harmonize even `![ab]->cannon_enc` checks? Well, one can say that since it goes all the way down to 1.0.2, then it might be not as appropriate... Side note then?

On another side note, function shamelessly breaks `const` contract, so it would be only appropriate to omit const qualifier [in next release]. But if considered inappropriate, then one can wonder if this contradiction should be resolved already now.

BTW, there is related "philosophical" question. While modification preserves original behaviour, one can wonder if it would appropriate to consider empty strings to be non-equal. Unless a and b are same. I mean empty string in one object can be viewed as non-equal to another. Rationale is that empty string would indicate an impossible value, and comparison of impossible values is meaningless. For example 1/x and 1/x^2 for x=0 are both impossible, but are they equal? The question is rhetorical.